src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs

```diff
@@ -272,24 +272,38 @@ bool ValidateApiLevels ()
 				(string.IsNullOrWhiteSpace (AndroidApiLevel) && string.IsNullOrWhiteSpace (TargetFrameworkVersion));
 
 			if (UseLatestAndroidPlatformSdk) {
-				AndroidApiLevel = GetMaxInstalledApiLevel ().ToString ();
-				SupportedApiLevel = GetMaxStableApiLevel ().ToString ();
-				int maxInstalled, maxSupported = 0;
-				if (int.TryParse (AndroidApiLevel, out maxInstalled) && int.TryParse (SupportedApiLevel, out maxSupported) && maxInstalled > maxSupported) {
+				int maxInstalled = GetMaxInstalledApiLevel ();
+				int maxSupported = GetMaxStableApiLevel ();
+				AndroidApiLevel = maxInstalled.ToString ();
+				SupportedApiLevel = maxSupported.ToString ();
+				if (maxInstalled > maxSupported) {
 					Log.LogDebugMessage ($"API Level {AndroidApiLevel} is greater than the maximum supported API level of {SupportedApiLevel}. " +
 						"Support for this API will be added in a future release.");
-					AndroidApiLevel = SupportedApiLevel;
 				}
 				if (!string.IsNullOrWhiteSpace (TargetFrameworkVersion)) {
 					var userSelected = MonoAndroidHelper.SupportedVersions.GetApiLevelFromFrameworkVersion (TargetFrameworkVersion);
 					// overwrite using user version only if it is 
 					// above the maxStableApi and a valid apiLevel.
 					if (userSelected != null && userSelected > maxSupported && userSelected <= maxInstalled) {
-						AndroidApiLevel = userSelected.ToString ();
-						SupportedApiLevel = userSelected.ToString ();
+						maxInstalled =
+							maxSupported = userSelected.Value;
+						AndroidApiLevel =
+							SupportedApiLevel = userSelected.ToString ();
+					}
+				}
+
+				for (int apiLevel = maxSupported; apiLevel >= MonoAndroidHelper.SupportedVersions.MinStableVersion.ApiLevel; apiLevel--) {
```

Comment:
Shouldn't this logic be in `GetTargetFrameworkVersionFromApiLevel()`, so that the "override `$(AndroidApiLevel)`" codepath also uses it? Or should the "override `$(AndroidApiLevel)`"  codepath *not* ensure that `$(TargetFrameworkVersion)` and `$(AndroidApiLevel)` are consistent with each other?